compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -997,21 +1046,37 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let ty = self.infcx.resolve_vars_if_possible(self.node_ty(var_hir_id));
 
         if !ty.has_significant_drop(self.tcx, self.tcx.param_env(closure_def_id.expect_local())) {
+            debug!("does not have significant drop");
             return None;
         }
 
         let Some(root_var_min_capture_list) = min_captures.and_then(|m| m.get(&var_hir_id)) else {
             // The upvar is mentioned within the closure but no path starting from it is
-            // used.
+            // used. This occurs when you have (e.g.)
+            //
+            // ```
+            // let x = move || {
+            //     let _ = y;
+            // });
+            // ```
+            debug!("no path starting from it is used");
+
 
             match closure_clause {
                 // Only migrate if closure is a move closure
-                hir::CaptureBy::Value => return Some(FxHashSet::default()),
```

Comment:
the error was this is empty? Little confusing what fixed it along side all the refactoring.

Reply:
Yes, that was the error. The refactoring was just me figuring out how to actually emit a good error at the end. =)